File: 《亿级流量Jαva高并发与网络编程实战》-源码/demo/ch08/thrift/nodejs/thriftClient.js
var thrift = require('thrift');
var StudentService = require('./StudentService.js');
var ttypes = require('./Student_types');

var transport = thrift.TFramedTransport;
var protocol = thrift.TBinaryProtocol;
var options = {transport: transport, protocol: protocol};
var connection = thrift.createConnection("127.0.0.1", 8888, options);

//创建客户端
client = thrift.createClient(StudentService, connection);

//处理异常
connection.on('error', function(err) {
    console.error(err);
});

client.addStudent("ww",25,function(err,result){
    console.log(result?"增加成功":"增加失败");
});

client.queryStudents(function(err,result){
    result.forEach(function(student,index){
        console.log("查询结果如下：")
        console.log(student.name + "," +student.age)
    });
});

File: 《亿级流量Jαva高并发与网络编程实战》-源码/demo/ch08/thrift/python/ThriftClient.py
#以python方法导入thrift相关类
from thrift.Thrift import TException
from thrift.protocol import TCompactProtocol, TBinaryProtocol
from thrift.transport import TSocket, TTransport
from genpy.thrift.generatecode import StudentService

tSocket = TSocket.TSocket("127.0.0.1", 8888)
tSocket.setTimeout(1000)
# 指定传输方式
transport = TTransport.TFramedTransport(tSocket)
# 指定传输格式
protocol = TBinaryProtocol.TBinaryProtocol(transport)
client = StudentService.Client(protocol)
transport.open()
try:
    print("Python RPC调用 queryStudents()方法")
    students = client.queryStudents()
    for student in students:
        print ("姓名:%s，年龄:%d" % (student.name , student.age) )

    print("Python RPC调用 addStudent()方法")
    result = client.addStudent("zs",23)
    if result:
        print("增加成功")
    transport.close()
except TException as e:
    print(e)